spring-jdbc/src/main/java/org/springframework/jdbc/datasource/DataSourceUtils.java

```diff
@@ -151,14 +148,13 @@ public static Connection doGetConnection(DataSource dataSource) throws SQLExcept
 	 * {@link DataSource#getConnection()} into an {@link IllegalStateException}.
 	 * @param dataSource the DataSource to obtain Connections from
 	 * @return a JDBC Connection from the given DataSource (never {@code null})
-	 * @throws SQLException if thrown by JDBC methods
-	 * @throws IllegalStateException if the DataSource returned a null value
+	 * @throws SQLException if thrown by JDBC methods or the DataSource returned a null value
 	 * @see DataSource#getConnection()
 	 */
 	private static Connection fetchConnection(DataSource dataSource) throws SQLException {
 		Connection con = dataSource.getConnection();
 		if (con == null) {
-			throw new IllegalStateException("DataSource returned null from getConnection(): " + dataSource);
+			throw new SQLException("DataSource returned null from getConnection(): " + dataSource);
```

Comment:
That change doesn't sound right to me. We shouldn't be throwing `SQLException` here.

Reply:
I did this change because the `CannotGetJdbcConnectionException` only accepts SQLException as cause parameter. 
So what do you think of relaxing the type constraint in `CannotGetJdbcConnectionException` a bit so that also IllegalStateException could be provided either via a constructor overload or a more relaxed type specified.

Reply:
So i would interpret this like the PR could be just closed.

Reply:
Ok i will adjust the code accordingly.